src/views/Main.js

```diff
@@ -0,0 +1,73 @@
+import { useEffect, useState } from 'react';
+import TypeDropdown from '../components/TypeDropdown';
+import Input from '../components/Input';
+import PokeCard from '../components/PokeCard/PokeCard';
+
+import {
+  fetchPokemon,
+  fetchPokemonByName,
+  fetchPokemonType,
+  fetchSearchedPokemon,
+  fetchSelectedPokemonType,
+} from '../services/pokemon';
+
+export default function Main() {
+  const [pokedex, setPokedex] = useState([]);
+  const [type, setType] = useState([]);
+  const [selectType, setSelectType] = useState('All');
+  const [searchBar, setSearchBar] = useState('');
+  const [loading, setLoading] = useState(true);
+  useEffect(() => {
+    const fetchData = async () => {
+      if (selectType === 'All') {
+        const data = await fetchPokemon();
+        setPokedex(data);
+        setLoading(false);
+      }
+    };
+    const fetchType = async () => {
+      const data = await fetchPokemonType();
+      setType(data);
+      setLoading(false);
+    };
+    fetchData();
+    fetchType();
+  }, [selectType]);
+
+  useEffect(() => {
```

Comment:
great working with the naming conventions-- makes reading your code clear and easy